pkgs/games/xonotic/default.nix

```diff
@@ -98,32 +101,46 @@ let
 
     # Xonotic needs to find libcurl.so at runtime for map downloads
     dontPatchELF = true;
-    postFixup = lib.optionalString withDedicated ''
-      patchelf --add-needed ${curl.out}/lib/libcurl.so $out/bin/xonotic-dedicated
-    '' + lib.optionalString withGLX ''
-      patchelf \
-          --add-needed ${curl.out}/lib/libcurl.so \
-          --add-needed ${libvorbis}/lib/libvorbisfile.so \
-          --add-needed ${libvorbis}/lib/libvorbis.so \
-          --add-needed ${libGL.out}/lib/libGL.so \
-          $out/bin/xonotic-glx
-    '' + lib.optionalString withSDL ''
-      patchelf \
+
+    commonPatchelfArgs = ''
           --add-needed ${curl.out}/lib/libcurl.so \
-          --add-needed ${libvorbis}/lib/libvorbisfile.so \
-          --add-needed ${libvorbis}/lib/libvorbis.so \
-          $out/bin/xonotic-sdl
+          --add-needed ${zlib.out}/lib/libz.so \
+          --add-needed $out/lib/libd0_blind_id.so \
     '';
+
+    commonClientPatchelfArgs= ''
+          --add-needed ${libvorbis.out}/lib/libvorbis.so \
+          --add-needed ${libvorbis.out}/lib/libvorbisfile.so \
+          --add-needed ${libvorbis.out}/lib/libvorbisenc.so \
+          --add-needed ${libtheora.out}/lib/libtheora.so \
+          --add-needed ${libtheora.out}/lib/libtheoraenc.so \
+          --add-needed ${libtheora.out}/lib/libtheoradec.so \
+          --add-needed ${libpng.out}/lib/libpng16.so \
+          --add-needed ${freetype.out}/lib/libfreetype.so \
+    '';
+
+    postFixup =
+      lib.optionalString withDedicated (''
+          patchelf '' + commonPatchelfArgs + ''
```

Comment:
Nitpicks:
Maybe change the indentation to two spaces more than the previous line?
Maybe use `${commonPatchelfArgs}` instead of `'' + commonPatchelfArgs + ''`?